### PR TITLE
Improve intellisense behavior for MSSQL kernel.

### PR DESCRIPTION
**Pull request by corivera**
Fixes https://github.com/dotnet/interactive/issues/1048

I was noticing some issues in the MSSQL intellisense where things wouldn't autocomplete unless there were extra spaces or blank lines in the cell. My fixes here were to handle newlines correctly using Environment.NewLine so that there wouldn't be offset spacing in the text, as well as incrementing the last character column number by 1 to address off-by-1 issues. That last one doesn't really make sense, since we don't have to increment the line number by 1, so that may be a bug in the SqlToolsService somewhere. Trying lots of different text to autocomplete showed no issues when incrementing the column number in this way, and there were definitely lots of autocomplete issues without doing it. I also cleaned up an unused return type, and fixed a potential array out-of-bounds error when parsing the text to autocomplete.

### CODE REVIEW

File: src/Microsoft.DotNet.Interactive.SqlServer/MsSqlServiceClient.cs [hunk 1]
```diff
@@ -158,10 +156,10 @@ public async Task<QueryExecuteSubsetResult> ExecuteQueryExecuteSubsetAsync(Query
 
         public async Task SendTextChangeNotificationAsync(Uri ownerUri, string newText, string oldText)
         {
-            var oldTextLines = oldText.Split('\n');
+            var oldTextLines = oldText.Split(Environment.NewLine);
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.SqlServer/MsSqlServiceClient.cs:159]: It's a common scenario that the compute and the editor are on different machines with different newlines, or that a notebook file was written with `\r\n` newlines but is being evaluated with `\n` newlines. So the split should handle both.

Comment (colombod) [reply src/Microsoft.DotNet.Interactive.SqlServer/MsSqlServiceClient.cs:159]: in ntoebook parsing we use this logic 
`s.Split('\n').Select(l => l.EndsWith('\r') ? l[0..^1] : l);`

Comment (corivera) [reply src/Microsoft.DotNet.Interactive.SqlServer/MsSqlServiceClient.cs]: Fixed with Diego's suggestion.

File: src/Microsoft.DotNet.Interactive.SqlServer/MsSqlServiceClient.cs [hunk 2]
```diff
@@ -158,10 +156,10 @@ public async Task<QueryExecuteSubsetResult> ExecuteQueryExecuteSubsetAsync(Query
 
         public async Task SendTextChangeNotificationAsync(Uri ownerUri, string newText, string oldText)
         {
-            var oldTextLines = oldText.Split('\n');
+            var oldTextLines = oldText.Split('\n').Select(text => text.EndsWith('\r') ? text[0..^1] : text).ToArray();
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.SqlServer/MsSqlServiceClient.cs:159]: You can simplify like this:

```csharp
text.Split(new []{ '\n', '\r' }, StringSplitOptions.RemoveEmptyEntries)
```

Comment (corivera) [reply src/Microsoft.DotNet.Interactive.SqlServer/MsSqlServiceClient.cs:159]: Does that still preserve blank lines? We need the original number of lines in the text in order to get the right document change ranges.

Comment (jonsequitur) [reply src/Microsoft.DotNet.Interactive.SqlServer/MsSqlServiceClient.cs:159]: It does if you don't specify `StringSplitOptions.RemoveEmptyEntries`.

Comment (corivera) [reply src/Microsoft.DotNet.Interactive.SqlServer/MsSqlServiceClient.cs:159]: Either way doesn't seem to work. When using this test string, "abc\r\ndef\r\n\r\nabc\r\nabcdef", the parsing returns 4 lines when specifying RemoveEmptyEntries, while it returns 9 when leaving out that option. We expect 5 lines given the Windows line endings in that string.

Comment (jonsequitur) [reply src/Microsoft.DotNet.Interactive.SqlServer/MsSqlServiceClient.cs:159]: Ah, righto.

File: src/Microsoft.DotNet.Interactive.SqlServer.Tests/MsSqlServiceClientTests.cs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            int oldTextLineCount = 5;
+            int oldTextLastCharacterNum = 6;
+            string newText = string.Join(lineEnding, "abc", "def");
+            var testUri = new Uri("untitled://test");
+
+            var docChange = MsSqlServiceClient.GetDocumentChangeForText(testUri, newText, oldText);
+
+            docChange.ContentChanges.Length
+                .Should()
+                .Equals(1);
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.SqlServer.Tests/MsSqlServiceClientTests.cs:37]: This is a common `FluentAssertions` gotcha. You should make this change throughout these. The `Equals` call here is `Object.Equals` which returns false but doesn't throw, so you'll get false positives.

```suggestion
                .Be(1);
```

**Review by jonsequitur — APPROVED** [review]

**Pull request merged**